conf/distro/include/mali.inc

```diff
@@ -0,0 +1,25 @@
+MALI_USERLAND_LIBARIES ?= "mali450-userland"
+
+# Helper function for overloading the default EGL/GLES implementation.
+# The Mali libraries provided by ARM are compatible with the Mesa headers
+# and it is safe to use with user space applications linked against Mesa.
+
+def get_mali_handler(d, target):
+    """  Overloading the default EGL/GLES implementation."""
+    features = d.getVar('MACHINE_FEATURES', True).split()
+
+    mali_libs = d.getVar('MALI_USERLAND_LIBARIES', True);
+
+    if(mali_libs):
+        mali_libs = mali_libs.split()
+
+    if 'mali450' in features and mali_libs:
+        provider = mali_libs[0]
+    else:
+        provider = "mesa"
```

Comment:
The patch is ready to merge, however, please review this line if it is ok with you. We always fall back to Mesa if no mali defined, since getting the PREFERRED_PROVIDER_virtual/egl here would cause an infinite recursions.
I think this is ok for RPB at the moment.